src/tiled/commanddatamodel.cpp

```diff
@@ -455,7 +453,9 @@ void CommandDataModel::setShortcut(const QModelIndex &index, const QKeySequence
 {
     if (index.row() < mCommands.size()) {
         mCommands[index.row()].shortcut = keySequence;
-        emit dataChanged(index, index);
+
+        QModelIndex shortcutIndex = this->index(index.row(), ShortcutColumn);
+        emit dataChanged(index, shortcutIndex);
```

Comment:
Not sure why you're not using shortcutIndex for the first argument here? It should be.